### PR TITLE
Update profile watchers and utilities to use Zowe SDK enhancements

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

⚠️ Depends on Zowe CLI PRs:
* zowe/zowe-cli#2313
* zowe/zowe-cli#2323
* zowe/zowe-cli#2360

## Release Notes

Milestone: 3.1.0

Changelog:
- Fixed an issue where editing a team config file or updating credentials in OS vault could trigger multiple events for a single action.
- Updated Zowe SDKs to `8.8.2` for technical currency.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by anaxceron — DISMISSED** [review]

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 1]
```diff
@@ -332,48 +321,38 @@ export class ProfilesUtils {
...
+            credMgrOverride: hasSecureCredentialManagerEnabled && (await this.setupDefaultCredentialManager()),
         });
-        return profileInfo;
     }
 
     public static async readConfigFromDisk(warnForMissingSchema?: boolean): Promise<void> {
         ZoweLogger.trace("ProfilesUtils.readConfigFromDisk called.");
         let rootPath: string;
-        const mProfileInfo = await ProfilesUtils.getProfileInfo();
+        this.mProfileInfo = await ProfilesUtils.setupProfileInfo();
```
Comment (zFernand0) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:338]: Should we use the name of the class (`ProfileUtils.`) instead of `this.` throughout static functions?

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 2]
```diff
@@ -332,48 +321,38 @@ export class ProfilesUtils {
...
         }
 
-        const profileInfo = new imperative.ProfileInfo("zowe", {});
-        const workspacePath = ZoweVsCodeExtension.workspaceRoot?.uri.fsPath;
-        // Trigger initialize() function of credential manager to throw an error early if failed to load
-        await profileInfo.readProfilesFromDisk({
-            homeDir: FileManagement.getZoweDir(),
-            projectDir: workspacePath ? FileManagement.getFullPath(workspacePath) : undefined,
+        return new imperative.ProfileInfo("zowe", {
+            credMgrOverride: hasSecureCredentialManagerEnabled && (await this.setupDefaultCredentialManager()),
```
Comment (zFernand0) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:331]: I like the `&&` 😋 
but I'm curious if we should convert this to a full ternary operator and fallback to `undefined` instead of `false` 🤔 

Technically, it shouldn't cause troubles since this sort of thing is handled by the `imperative.ProfileCredentials`
`await CredentialManagerFactory.initialize({ service: null, ...this.mCredMgrOverride || {} });`

I'm curious how some static analysis tools will react to us possibly setting `credMgfOverride: false` 😋

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 3]
```diff
@@ -308,10 +297,10 @@ export class ProfilesUtils {
     }
 
     /**
-     * Creates an instance of ProfileInfo and calls `readProfilesFromDisk` to load profiles.
+     * Creates an instance of ProfileInfo with the configured credential manager override.
      * @returns An instance of `ProfileInfo`, or `undefined` if there was an error.
      */
-    public static async getProfileInfo(): Promise<imperative.ProfileInfo> {
+    public static async setupProfileInfo(): Promise<imperative.ProfileInfo> {
```
Comment (zFernand0) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:303]: This seems like a good change. I'm curious how many PRs may have conflicts after this 😅 

In theory, not many since this is different from the ProfilesCache.getProfileInfo 😋 

I would love it if this PR was merged ASAP 🙏

**Review by zFernand0 — COMMENTED** [review]
changes make sense to me 😋 

---

will approve after testing! 😋 

---

I know some of these lint issues weren't introduced by this PR.
However, if it's too much trouble, could we address them here?
https://github.com/zowe/zowe-explorer-vscode/actions/runs/11855312035/job/33039267040?pr=3296#step:6:129

File: packages/zowe-explorer/src/trees/shared/SharedInit.ts [hunk 4]
```diff
@@ -338,40 +318,36 @@ export class SharedInit {
...
-                Constants.SAVED_PROFILE_CONTENTS.set(uri.fsPath, newProfileContents);
-                void SharedActions.refreshAll();
-                ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.UPDATE);
-            });
+            watcher.onDidCreate(
+                SharedUtils.debounce(() => {
+                    ZoweLogger.info(vscode.l10n.t("Team config file created, refreshing Zowe Explorer."));
+                    void SharedActions.refreshAll();
+                    ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.CREATE);
+                }, 100) // eslint-disable-line no-magic-numbers
```
Comment (zFernand0) [inline packages/zowe-explorer/src/trees/shared/SharedInit.ts:326]: Should we make the `delay` optional and default it to 100?

File: packages/zowe-explorer/src/trees/shared/SharedInit.ts [hunk 5]
```diff
@@ -338,40 +318,36 @@ export class SharedInit {
...
-            (imperative.EventProcessor.prototype as any).emitZoweEvent = function (eventName: string): void {
-                SharedInit.emitZoweEventHook(this, eventName);
-            };
-
-            const zoweWatcher = imperative.EventOperator.getWatcher().subscribeUser(
-                imperative.ZoweUserEvents.ON_VAULT_CHANGED,
-                SharedInit.onVaultChanged
-            );
+            const zoweWatcher = imperative.EventOperator.getWatcher().subscribeUser(imperative.ZoweUserEvents.ON_VAULT_CHANGED, async () => {
+                ZoweLogger.info(vscode.l10n.t("Changes in the credential vault detected, refreshing Zowe Explorer."));
```
Comment (zFernand0) [inline packages/zowe-explorer/src/trees/shared/SharedInit.ts:346]: I'm probably missing something, but unless I write my own nodejs script (that calls the `save` function from the config.api.secure properties) I don't see how this event is getting triggered 🤯

And even with a custom script, this event is not captured in debug mode (it is only captured in a VSIX installation) 😕 

Because of this, running multiple instances of this PR may cause credentials to not be updated properly between multiple windows of Zowe Explorer 😢

**Review by zFernand0 — COMMENTED** [review]
TL;DR: LGTM! 😋 
---


The changes make the behavior a lot better since we don't refresh ZE as many times now, but I'm wondering if we are refreshing it too little now 😋 

I'll like to discuss a few things before approving though.

**Review by zFernand0 — CHANGES_REQUESTED** [review]
One more thing, and sorry for being picky.
There are changed to l10n files coming out of the prepublish script
- `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Review by anaxceron — DISMISSED** [review]

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts (hunk 2 above)
Comment (zFernand0) [reply packages/zowe-explorer/src/utils/ProfilesUtils.ts]: After some further discussion and testing, this was a problem with my `.env` file where I had `ZOWE_CLI_HOME` set to my current `.dist`. That meant that zowe events needed to happen there in order for ZE to capture them 😋 

Thanks @t1m0thyj

Comment (zFernand0) [reply packages/zowe-explorer/src/utils/ProfilesUtils.ts]: thanks for addressing this 🙏

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 

Thanks for addressing my dumb questions 😅

**Review by JillieBeanSim — DISMISSED** [review]
LGTM! thanks @t1m0thyj

**Review by JillieBeanSim — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

File: packages/zowe-explorer/src/trees/shared/SharedUtils.ts [hunk 6]
```diff
@@ -333,4 +333,19 @@ export class SharedUtils {
             });
         }
     }
+
+    /**
+     * Debounces an event handler to prevent duplicate triggers.
+     * @param callback Event handler callback
+     * @param delay Number of milliseconds to delay
+     */
+    public static debounce<T extends (...args: any[]) => void>(callback: T, delay: number): (...args: Parameters<T>) => void {
```
Comment (traeok) [inline packages/zowe-explorer/src/trees/shared/SharedUtils.ts:342]: Should we add patch coverage for this function? I see its being mocked for one of the tests, but we might want to validate the debounce behavior just in case its used elsewhere 😋

**Review by traeok — COMMENTED** [review]
Thanks for the updates and refactoring Timothy - I had a question about testing `SharedUtils.debounce`, but I tested the config watcher and everything LGTM
